addons/mission/functions/fnc_markBuildings.sqf

```diff
@@ -0,0 +1,47 @@
+#include "..\script_component.hpp"
+/*
+ * Author: Mike
+ * Creates map markers for building outlines with an optional filter for only objects that inherit from buildings.
+ * Function should not be ran at any point after mission start.
+ * This is mainly designed around houses and custom structures being marked on the map, if using a layer ensure only buildings or walls are inside it.
+ *
+ * Call from initServer.sqf on mission start.
+ *
+ * Arguments:
+ * 0: Objects <ARRAY>
+ * 1: Filter <BOOL> (default: true)
+ *
+ * Return Value:
+ * None
+ *
+ * Examples:
+ * [[MyObject, MyObject2]] call MFUNC(markBuildings);
+ * [(getMissionLayerEntities "Test Layer" select 0)] call MFUNC(markBuildings);
+*/
+
+params ["_objectArray", ["_filter", true]];
+
+if (count _objectArray == 0) exitWith {
+    WARNING("Object Array is empty");
+};
+
+// Filter doesn't catch everything, Arma has so many weird objects inheriting from Building, House_F etc.
+if (_filter) then {
+    _objectArray = _objectArray select {_x isKindOf "Building"};
+};
+
+{
+    private _bb = boundingBoxReal _x params ["_posRelative1", "_posRelative2"];
+    private _width = abs (_posRelative2 select 0) - (_posRelative1 select 0);
+    private _height = abs (_posRelative2 select 1) - (_posRelative1 select 1);
+
+    // Marker name has to be unique.
+    private _namePos = position _x params ["_xPos", "_yPos", "_zPos"];
+    private _markerName = format ["%1%2%3", _xPos, _yPos, _zPos];
```

Comment:
```suggestion
    private _markerName = format ["%1_%2%3%1", QUOTE(ADDON), _xPos, _yPos, _zPos];
```
I'd do something like this personally, just so a cleanup could be more easily done if needed for some reason.

idk if something is better than `QUOTE(ADDON)` here though